shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -133,6 +133,34 @@ FLUTTER_EXPORT
  */
 - (void)destroyContext;
 
+/**
+ * Ensures that Flutter will generate a semantics tree.
+ *
+ * This is enabled by default if certain accessibility services are turned on by
+ * the user, or when using a Simulator. This method allows a user to turn
+ * semantics on when they would not ordinarily be generated and the performance
+ * overhead is not a concern, e.g. for UI testing. Note that semantics should
+ * never be programatically turned off, as it would potentially disable
+ * accessibility services an end user has requested.
+ *
+ * This method must only be called after launching the engine via
+ * `-runWithEntrypoint:` or `-runWithEntryPoint:libraryURI`.
+ */
+- (void)ensureSemanticsEnabled;
+
+/**
+ * Registers a callback that will be called once when semantics are available.
+ *
+ * If semantics are disabled, this callback will be stored but will not fire
+ * until they are enabled and the tree has been updated. Consider calling
+ * `-ensureSemanticsEnabled` first.
+ *
+ * Multiple calls to this method will replace the callback.
```

Comment:
I would propose (weakly) holding all of the callbacks instead of overriding, and make this API in the form of register/unregisterSomeThing.

As both a downstream user of the callback and a framework provider, EarlGrey will register this callback to ensure Flutter is synchronized, but end user may also be interested in registering the callback and causing unexpected result at either side. Though there is hacky way to get around this, it's always best to act as a 'nice' user.

I think register/deregister methods can be put either here or downstream FlutterViewController.

Reply:
That makes sense.  I was hoping I could avoid the complexity involved there but you're right.

Should it still be a fire once and automatically unregister?

Reply:
Maybe we could use NSNotificationCenter for this.

Reply:
Vote for NSNotificationCenter. One-shot subscription could be done by user-side block so I think it should be fine.